### PR TITLE
fix: encode dask arrays in chunks and encode arrays of shape (1,)

### PR DESCRIPTION
**Pull request by falkephi**

### CODE REVIEW

File: opendap_protocol/protocol.py [hunk 1]
```diff
@@ -552,3 +552,12 @@ def meets_constraint(constraint_expr, data_path):
             return True
 
     return False
+
+
+def set_dask_encoding_chunk_size(chunk_size: int):
```
Comment (gabrielaznar) [inline opendap_protocol/protocol.py:557]: do we need this method? the class attributes could be changed just by importing.

File: opendap_protocol/protocol.py [hunk 2]
```diff
@@ -552,3 +552,12 @@ def meets_constraint(constraint_expr, data_path):
...
+    """Set the maximum chunk size used to encode ``dask.Array``s to XDR.
+
+    :param chunk_size: (int) Encoding chunk size in Bytes
+    :returns: None
+    """
+    Config.DASK_ENCODE_CHUNK_SIZE = chunk_size
```
[the remark this answers is not in the document]
Comment (falkephi) [reply opendap_protocol/protocol.py:563]: I propose to cast it to ``int``, which will raise an error on all non-numeric values plus we enforce positive values.

File: tests/test_all.py [hunk 3]
```diff
@@ -183,32 +193,107 @@ def test_complete_dap_response():
...
-    assert y.parent == dataset
+    assert x_dap.parent == dataset
+    assert y_dap.parent == dataset
+
+
+@pytest.mark.parametrize('x,y,z',
+                         test_arrays,
+                         ids=['numpy.array', 'dask.array'])
+def test_sliced_dap_response(x, y, z):
+    """Test if slicing of a DAP dataset works as expected, including a
```
[the remark this answers is not in the document]
Comment (falkephi) [reply tests/test_all.py:235]: I don't remember. Wrote something useful now.

File: opendap_protocol/protocol.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (nestabur) [reply opendap_protocol/protocol.py]: casting to int will only help with the non-numeric values and None, but won't enforce positive values.

[the remark this answers is not in the document]
Comment (falkephi) [reply opendap_protocol/protocol.py]: That formulation above was misleading. We will cast to ``int`` and in addition enforce positive values as implemented in the most recent commits.

**Review by nestabur — APPROVED** [review]

**Pull request merged**